cmd/state-svc/internal/rtwatcher/watcher.go

```diff
@@ -0,0 +1,116 @@
+package rtwatcher
+
+import (
+	"encoding/json"
+	"os"
+	"runtime/debug"
+	"strconv"
+	"time"
+
+	"github.com/ActiveState/cli/internal/analytics/client/sync"
+	"github.com/ActiveState/cli/internal/analytics/constants"
+	"github.com/ActiveState/cli/internal/analytics/dimensions"
+	"github.com/ActiveState/cli/internal/config"
+	constants2 "github.com/ActiveState/cli/internal/constants"
+	"github.com/ActiveState/cli/internal/errs"
+	"github.com/ActiveState/cli/internal/logging"
+	"github.com/ActiveState/cli/internal/runbits/panics"
+)
+
+const defaultInterval = 1 * time.Minute
+const CfgKey = "runtime-watchers"
+
+type Watcher struct {
+	an       *sync.Client
+	cfg      *config.Instance
+	watching []entry
+	stop     chan struct{}
+}
+
+func New(cfg *config.Instance, an *sync.Client) *Watcher {
+	w := &Watcher{an: an, stop: make(chan struct{}), cfg: cfg}
+
+	if watchersJson := w.cfg.GetString(CfgKey); watchersJson != "" {
```

Comment:
I'm not sure if we care enough but if a savvy user cleaned their config and restarted the state service we would not be able to track the heartbeat anymore. This could also potentially happen if something went wrong with an update. While it wouldn't provide us _much_ more protection we could use the `persist` directory that contains the `uniqid` file. At least this way we aren't providing users with a path to undermine the heartbeat tracking.

Might be better for a follow-up story though.

Reply:
Not worth it. The runtime tracking is a best effort kind of thing. There's lots of things people could do to circumvent it or limit its use, but we care about the 99% of the users who don't mess about rather than the 1% that do.

Reply:
Fair enough. Though if I recall this data is going to be used to help determine billing? Or what tier a user/org should be on. If that's the case this may be something we have to revisit.

Reply:
Yeah but the data is only used to manually determine the bill. I gave the same concerns you did when we discussed this, the above take was effectively the answer I received.